### PR TITLE
Implementing professional experience create

### PR DESCRIPTION
**Pull request by nehamamandelbaum**
This PR completes issue #567, which was to implement the create method of the professional experiences in the user interface (outside of the admin interface). 

How the UI looks: 

Captura de Tela 2023-10-25 às 12 19 36

Captura de Tela 2023-10-25 às 11 08 11

### CODE REVIEW

File: app/controllers/professional_experiences_controller.rb [hunk 1]
```diff
@@ -4,4 +4,45 @@ class ProfessionalExperiencesController < ApplicationController
...
+  def professional_experience_params
+    params.require(:professional_experience).permit(:company, :position, :description, :responsabilities, :start_date,
+                                                    :end_date)
+  end
+
+  def flash_errors(scope)
+    flash.now[:alert] = "#{alert_message(scope)} #{error_message}"
+  end
+
+  def alert_message(scope)
```
[the remark this answers is not in the document]
Comment (nehamamandelbaum) [reply app/controllers/professional_experiences_controller.rb]: Realmente, vou usar nas outras controllers também

File: app/controllers/professional_experiences_controller.rb [hunk 2]
```diff
@@ -4,4 +4,37 @@ class ProfessionalExperiencesController < ApplicationController
...
     @professional_experiences = current_user.professional_experiences.page(params[:page]).per(params[:per])
   end
+
+  def new
+    @professional_experience = ProfessionalExperience.new
+  end
+
+  def create
+    @professional_experience = ProfessionalExperience.new(professional_experience_params)
+    @professional_experience.user = current_user
```
Comment (adbatista) [inline app/controllers/professional_experiences_controller.rb:14]:
```suggestion
    @professional_experience = current_user.professional_experiences.new(professional_experience_params)
```

File: spec/requests/professional_experiences_spec.rb [hunk 3]
```diff
@@ -42,4 +42,69 @@
...
+  end
+
+  describe 'POST #create' do
+    context 'when user is signed in' do
+      let(:user) { create(:user) }
+      before { sign_in user }
+
+      context 'with valid params' do
+        let(:valid_params) do
+          { professional_experience: { company: 'Codeminer42', position: 'Software engineer', description: 'some description here', responsibilities: 'Ruby on Rails', start_date: '11/2021',
```
Comment (adbatista) [inline spec/requests/professional_experiences_spec.rb:74]: you can use the method attributes_for of FactoryBot

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -14,24 +14,22 @@ class ApplicationController < ActionController::Base
...
-    if cannot?(:read, ActiveAdmin)
-      redirect_to root_path, alert: I18n.t('devise.failure.access_denied')
-    end
+    return unless cannot?(:read, ActiveAdmin)
+
+    redirect_to root_path, alert: I18n.t('devise.failure.access_denied')
   end
 
-  def after_sign_in_path_for(user)
+  def after_sign_in_path_for(_user)
```
Comment (adbatista) [inline app/controllers/application_controller.rb:32]: ❤️

**Review by adbatista — APPROVED** [review]

**Pull request merged**